test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VSRestoreSettingsUtilityTests.cs

```diff
@@ -0,0 +1,151 @@
+
+                //Assert
+                Assert.True(
+                       Enumerable.SequenceEqual(expectedRestoreSources.OrderBy(t => t), actualSources.Select(e => e.Source).OrderBy(t => t)),
+                       "expected: " + string.Join(",", expectedRestoreSources.ToArray()) + "\nactual: " + string.Join(",", actualSources.Select(e => e.Source).ToArray()));
+            }
+        }
+
+        public static IEnumerable<object[]> GetVSRestoreSettingsUtilities_RestoreSourceData()
+        {
+            yield return new object[] {
+                new string[] { @"C:\source1" },
+                new string[] { @"C:\source1" }
+            };
+
+            yield return new object[]
+            {
+                new string[] { @"Clear" },
+                new string[] { }
+            };
+
+            yield return new object[]
+            {
+                new string[] { @"Clear", "RestoreAdditionalProjectSources", @"C:\additionalSource" },
+                new string[] { @"C:\additionalSource" }
+            };
+
+            yield return new object[] {
+                new string[] { @"C:\source1", "RestoreAdditionalProjectSources",@"C:\additionalSource" },
+                new string[] { @"C:\source1" ,@"C:\additionalSource" }
+            };
+
+            yield return new object[]
+            {
+                new string[] { "RestoreAdditionalProjectSources", @"C:\additionalSource" },
+                new string[] { NuGetConstants.V3FeedUrl, @"C:\additionalSource" }
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetVSRestoreSettingsUtilities_FallbackFolderData))]
+        public void VSRestoreSettingsUtilities_FallbackFolder(string[] fallbackFolders, string[] expectedFallbackFolders)
+        {
+            using (var mockBaseDirectory = TestDirectory.Create())
+            {
+                //Set Up
+                var spec = new PackageSpec();
+                spec.RestoreMetadata = new ProjectRestoreMetadata();
+                spec.RestoreMetadata.ProjectPath = @"C:\project\projectPath.csproj";
+                spec.RestoreMetadata.FallbackFolders = fallbackFolders.ToList();
+                var settings = new Settings(mockBaseDirectory);
+                settings.SetValue(ConfigurationConstants.FallbackPackageFolders, "defaultFallback", @"C:\defaultFallback");
+
+                //Act
+                var actualFallbackFolders = VSRestoreSettingsUtilities.GetFallbackFolders(settings, spec);
+
+                //Assert
+                Assert.True(
+                       Enumerable.SequenceEqual(expectedFallbackFolders.OrderBy(t => t), actualFallbackFolders.OrderBy(t => t)),
+                       "expected: " + string.Join(",", expectedFallbackFolders.ToArray()) + "\nactual: " + string.Join(",", actualFallbackFolders.ToArray()));
+            }
+        }
+
+        public static IEnumerable<object[]> GetVSRestoreSettingsUtilities_FallbackFolderData()
```

Comment:
why is this public? Consider moving it to a TestUtility class if this is going to be reused somewhere else.